azure/durable_functions/tasks/task_all.py

```diff
@@ -62,6 +67,9 @@ def task_all(tasks: List[Task]):
         results = []
         end_time = None
 
+    if replay_schema == ReplaySchema.V2:
+        actions = WhenAllAction(actions)
```

Comment:
Isn't `WhenAllAction` a single action, and actions expecting an `Action[]`?

Reply:
Your asking why the single action `WhenAllAction` is assigned to a variable named `actions`, right? Yeah that was a weird choice on my end. Looking back, it's not very readable.

The reason is that, in the case of `V1` replays, `actions` is a list of action objects. Since we already use this variable in constructing the TaskSet below, I figured I would just override its contents for V2 and keep the remaining logic intact. 

In general, a lot of this PR is designed to modify as little of the existing logic as possible, as I worried that too big of a refactor would risk introducing bugs. That being said, in this case we're talking about a specific variable name, so I don't mind just replacing the name with something like "action_payload" which works for V1 and V2.